examples/compare_coherency_methods.py

```diff
@@ -0,0 +1,593 @@
+# Taking the absolute CaCoh value shows us the magnitude of this interaction to
+# be ~0.9, which is also seen when first projecting this information to the
+# imaginary axis.
+#
+# This distinction is why connectivity methods utilising information from both
+# real and imaginary components (Cohy, Coh, CaCoh) capture both zero and
+# non-zero time-lag interactions, whereas methods using only the imaginary
+# component (ImCoh, MIC, MIM) capture only non-zero time-lag interactions.
+#
+# The ability to capture these different interactions is not a feature specific
+# to multivariate connectivity methods, as shown below for the bivariate
+# methods Coh and ImCoh.
+
+# %%
+
+# Compute Coh & ImCoh
+(coh, imcoh) = spectral_connectivity_epochs(
+    data, method=["coh", "imcoh"], indices=bivar_indices, sfreq=100, fmin=3, fmax=35
+)
+
+coh_mean = np.mean(coh.get_data(), axis=0)
+imcoh_mean = np.mean(np.abs(imcoh.get_data()), axis=0)
+
+coh_mean_subbed = coh_mean - np.mean(coh_mean)
+imcoh_mean_subbed = imcoh_mean - np.mean(imcoh_mean)
+
+# Plot Coh & ImCoh
+fig, axis = plt.subplots(1, 1)
+axis.plot(coh.freqs, coh_mean_subbed, linewidth=2, label="Coh")
+axis.plot(imcoh.freqs, imcoh_mean_subbed, linewidth=2, label="ImCoh", linestyle="--")
+axis.set_xlabel("Frequency (Hz)")
+axis.set_ylabel("Mean-corrected connectivity (A.U.)")
+axis.annotate("Non-zero\ntime-lag\ninteraction", xy=(13, 0.25))
+axis.annotate("Zero\ntime-lag\ninteraction", xy=(25, 0.25))
+axis.legend(loc="upper left")
+fig.suptitle("Coh vs. ImCoh\nNon-zero & zero time-lags")
+
+###############################################################################
+# When different coherency-based methods are most appropriate
+# -----------------------------------------------------------
+#
+# With this information, we can define situations under which these different
+# approaches are most appropriate.
+#
+# |
+#
+# **In situations where non-physiological zero time-lag interactions are
+# assumed, methods based on only the imaginary part of coherency (ImCoh, MIC,
+# MIM) should be used.** Examples of situations include:
+#
+# - Connectivity between channels of a single modality.
+# - Connectivity between channels of different modalities where the same
+#   reference is used.
+#
+# Note that this applies not only to sensor-space signals, but also to
+# source-space signals where remnants of these non-physiological interactions
+# may remain even after source reconstruction.
+#
+# |
+#
+# **In situations where non-physiological zero time-lag interactions are not
+# assumed, methods based on real and imaginary parts of coherency (Cohy, Coh,
+# CaCoh) should be used.** Examples of situations include:
+#
+# - Connectivity between channels of different modalities where different
+#   references are used.
+#
+# |
+#
+# Equally, it is important to avoid methods based on only the imaginary part of
+# coherency when non-physiological zero time-lag interactions are absent. There
+# are two key reasons:
```

Comment:
Perhaps this can be rephrased as there are multiple negatives in here "avoid" "non-*" "absent", so it's sort of a mental struggle to capture the true meaning :p 

Perhaps something like:

"When there there are no non-physiological zero time-lag interactions, then one should not use imaginary part of coherncy"? 

Idk if that is even right though.

Reply:
Agreed!

https://github.com/mne-tools/mne-connectivity/blob/8652da468637d12b5ae7f9c3ee580cd76a55b3d4/examples/compare_coherency_methods.py#L383-L385